### PR TITLE
Drop support for Kubernetes < 1.15

### LINKED ISSUE #3170: Cleanup v1.10 - v1.14 related code
Since quite some time our testing for v1.10 and v1.11 in the README.md states:

```
[1] Version is technically supported but no longer actively tested. Regressions will go unnoticed.
```

I believe we can proceed with clean up of the handling for these versions for the following reasons:
- we you do a change in g/g or any other component you (as PR author) have to consider whether your change will work (whether the corresponding API version is supported)  in v1.10 or v1.11
- v1.10 and v1.11 are pretty old and IMO it does not bring much value to have these versions listed as `technically supported` when we don't do any testing against them. ` technically supported but no longer actively tested` is a vague statement

/kind cleanup
/priority normal

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area open-source dev-productivity
/kind cleanup

**What this PR does / why we need it**:
This PR drops all code related to Kubernetes versions < `1.15` as suggested in #3170. Since quite a long time we don't test or maintain these versions anymore, and given that they are quite old and also no longer maintained by the Kubernetes community, it doesn't make sense to carry to code along any longer.

This means that the minimum Kubernetes version for seed clusters is also raised to `1.15` (from `1.11`).

The `.spec.kubernetes.kubeControllerManager.horizontalPodAutoscaler.{up,down}scaleDelay` fields have been dropped because they are no longer meaningful.

**Which issue(s) this PR fixes**:
Fixes #3170

**Release note**:

```action operator
⚠️ Gardener does no longer support shoot clusters with Kubernetes versions < 1.15. With this change, the `.spec.kubernetes.kubeControllerManager.horizontalPodAutoscaler.{up,down}scaleDelay` fields have been dropped because they are no longer meaningful. Make sure to upgrade all existing clusters before upgrading to this Gardener version.
```
```action operator
⚠️ The minimum Kubernetes version for seed clusters has been raised from `v1.11` to `v1.15`. Make sure that all your registered seed clusters meet this requirement before upgrading to this Gardener version.
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,18 +39,17 @@ Gardener takes part in the [Certified Kubernetes Conformance Program](https://ww
...
+| **Packet** | N/A | N/A | N/A | N/A | N/A | N/A |
+| **vSphere** | N/A | N/A | N/A | N/A | N/A | N/A |
 
 [1] Version is technically supported but no longer actively tested. Regressions will go unnoticed.<br>
-[2] Conformance tests are still executed and validated, unfortunately [no longer shown in TestGrid](https://github.com/kubernetes/test-infra/pull/18509#issuecomment-668204180).
```
Comment (ialidzhikov) [inline README.md:53]: Hm, why 1.15 is now switched to the category [1] (`Version is technically supported but no longer actively tested. [...]`)? Currently we still actively test against 1.15, so it should still point to [2] (`Conformance tests are still executed and validated, [...]`).

Comment (rfranzke) [reply README.md:53]: fixed

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 2]
```diff
@@ -640,20 +636,13 @@ type HorizontalPodAutoscalerConfig struct {
...
 	// +optional
 	Tolerance *float64 `json:"tolerance,omitempty" protobuf:"fixed64,6,opt,name=tolerance"`
-	// The period since last upscale, before another upscale can be performed in horizontal pod autoscaler.
-	// +optional
-	UpscaleDelay *metav1.Duration `json:"upscaleDelay,omitempty" protobuf:"bytes,7,opt,name=upscaleDelay"`
 }
 
 const (
-	// DefaultHPADownscaleDelay is a constant for the default HPA downscale delay for a Shoot cluster.
-	DefaultHPADownscaleDelay = 15 * time.Minute
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/types_shoot.go:650]: Can you also remove the corresponding constant from v1alpha1?

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 3]
```diff
@@ -640,20 +636,13 @@ type HorizontalPodAutoscalerConfig struct {
...
 	// DefaultHPASyncPeriod is a constant for the default HPA sync period for a Shoot cluster.
 	DefaultHPASyncPeriod = 30 * time.Second
 	// DefaultHPATolerance is a constant for the default HPA tolerance for a Shoot cluster.
 	DefaultHPATolerance = 0.1
-	// DefaultHPAUpscaleDelay is for the default HPA upscale delay for a Shoot cluster.
-	DefaultHPAUpscaleDelay = 1 * time.Minute
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/types_shoot.go:656]: Can you also remove the corresponding constant from v1alpha1?

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/component_test.go [hunk 4]
```diff
@@ -132,91 +132,6 @@ ExecStart=/opt/bin/health-monitor-kubelet`),
 			))
 		},
 
-		Entry(
-			"kubernetes 1.10, w/ docker",
-			"1.10.0",
-			extensionsv1alpha1.CRINameDocker,
-			execStartPreLess117,
-			kubeletFlagsDocker(extensionsv1alpha1.CRINameDocker, true, true, true, true, true, true),
```
Comment (ialidzhikov) [inline pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/component_test.go:140]: You can also adapt the signature of `kubeletFlagsDocker` func. Params such as `featureGatePodPriority bool`, `featureGateSupportPodPidsLimit bool`, `cadvisorPort bool` are no longer needed.

File: pkg/operation/botanist/component/kubescheduler/kube_scheduler.go [hunk 5]
```diff
@@ -329,31 +329,9 @@ func (k *kubeScheduler) emptyManagedResourceSecret() *corev1.Secret {
 }
 
 func (k *kubeScheduler) reconcileShootResources(ctx context.Context) error {
-	if versionConstraintK8sEqual113.Check(k.version) {
-		data, err := k.computeShootResourcesData()
```
Comment (ialidzhikov) [inline pkg/operation/botanist/component/kubescheduler/kube_scheduler.go:333]: `computeShootResourcesData` func is no longer used, i.e can be removed now.

**Review by ialidzhikov — DISMISSED** [review]
/lgtm
/squash

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**